tree/tree.go

```diff
@@ -0,0 +1,32 @@
+package tree
+
+import (
+	"os"
+	"strings"
+)
+
+type Tree struct {
+	Name string
+	Address string
+	Dirs []*Tree
+	Files []string
+}
+
+func NewTree(address string) (*Tree, error) {
+	var t Tree
+	tmpS := strings.Split(address, "/")
+	t.Name = tmpS[len(tmpS)-1]
+	dirEntries, err := os.ReadDir(address)
+	if err != nil {
+		return nil, err
+	}
+	for _, dirEntry := range dirEntries {
```

Comment:
You can optimize your tree building by running different branches in different routines. However, this needs to be done carefully, because it could easily eat up all your memory. Consider using a worker pool.